pycbc/inference/models/base.py

```diff
@@ -563,7 +563,16 @@ def logprior(self):
     def _logprior(self):
         """Calculates the log prior at the current parameters."""
         logj = self.logjacobian
-        logp = self.prior_distribution(**self.current_params) + logj
+        # allow using constraint in the config which sets
```

Comment:
This seems to leak the interface of the marginalization parameters so that it is coded in multiple places. Would it be better to have the DistMarg class perhaps just override this function instead?

Reply:
Can you explain more?